tdrs-frontend/src/components/FileUpload/FileUpload.jsx

```diff
@@ -12,45 +14,108 @@ function FileUpload({ section }) {
 
-  const formattedSectionName = file.section
+  const formattedSectionName = selectedFile.section
     .split(' ')
     .map((word) => word.toLowerCase())
     .join('-')
 
-  const fileName = file?.fileName
+  const fileName = selectedFile?.fileName
   const hasUploadedFile = Boolean(fileName)
 
   const ariaDescription = hasUploadedFile
-    ? `Selected File ${file?.fileName}. To change the selected file, click this button.`
+    ? `Selected File ${selectedFile?.fileName}. To change the selected file, click this button.`
     : `Drag file here or choose from folder.`
 
-  const uploadFile = ({ target }) => {
-    dispatch(clearError({ section: target.name }))
-    dispatch(
-      upload({
-        file: target.files[0],
-        section: target.name,
-      })
-    )
+  const inputRef = useRef(null)
+
+  const validateAndUploadFile = (event) => {
+    const { name } = event.target
+    const file = event.target.files[0]
+
+    // Clear existing errors and the current file in the state if the user is re-uploading
+    dispatch(clearError({ section: name }))
+    dispatch(clearFile({ section: name }))
+
+    // Get the the first 4 bytes of the file with which to check file signatures
+    const blob = file.slice(0, 4)
+
+    const input = inputRef.current
+    const dropTarget = inputRef.current.parentNode
+
+    const filereader = new FileReader()
+
+    filereader.onloadend = (evt) => {
+      if (!evt.target.error) {
+        // Read in the file blob "headers: and create a hex string signature
+        const uint = new Uint8Array(evt.target.result)
+        const bytes = []
+        uint.forEach((byte) => {
+          bytes.push(byte.toString(16))
+        })
+        const header = bytes.join('')
+
+        switch (header) {
+          // For some reason, fileType.fromBlob won't detect image/png;
+          // Account for this by checking for png and some other
+          // file signatures manually.
+          case '89504e47':
+          case '47494638':
+          case 'ffd8ffe0':
+          case 'ffd8ffe1':
+          case 'ffd8ffe2':
+          case 'ffd8ffe3':
+          case 'ffd8ffe8':
+            // reject the file and create an error message
+            createFileInputErrorState(input, dropTarget)
+            return
+          default:
+            break
+        }
+
+        // file-type should detect and return values for most other
+        // known binary files
+        fileType.fromBlob(blob).then((res) => {
+          // res should be undefined for non-binary files
+          if (res) {
+            // reject the file and create an error message
+
+            createFileInputErrorState(input, dropTarget)
+          }
+        })
+
+        // At this point we can reasonably conclude the file is a text file.
+        // Add the file to the redux state
+        dispatch(
+          upload({
+            section: name,
+            file,
+          })
+        )
+      }
+    }
```

Comment:
This goes through and tries to detect if an uploaded file is a binary, and if so it is removed from the UI and an error state is triggered.